packages/core/src/indexing/local-index-api.ts

```diff
@@ -28,16 +29,30 @@ async function _getIndexModelArgs(
     throw new Error(`Cannot index ${modelStreamId.toString()}, it is not a Model StreamID`)
   }
 
+  let opts: IndexModelArgs = {
+    model: modelStreamId,
+  }
+
   if (modelStreamId.type == Model.STREAM_TYPE_ID) {
     const modelState = await repository.load(modelStreamId, {})
     const content = modelState.state.next?.content ?? modelState.state.content
     Model.assertVersionValid(content, 'major')
     if (content.relations) {
-      return { model: modelStreamId, relations: content.relations }
+      opts = {
```

Comment:
there's no need to make a new opts object and copy the existing fields over.  This assignment to opts could instead just be `opts.relations = content.relations`

Reply:
ping

ditto below with indices

Reply:
Can't do that with this specific type, since all fields are readonly. Or I can mark the fields not readonly.

Reply:
oh yeah, interesting. I don't usually see the `readonly` keyword used within interface types, usually only within classes.  Curious if @PaulLeCam has thoughts on best practices here

Reply:
Same, I've only used `readonly` for classes getters, not for plain objects, but I think it's good to have immutable objects rather than having to clone them every time there are possible mutations.

Reply:
No I really mean immutable. Objects that are passed as arguments to functions shouldn't be mutated IMO so it's good if they are clearly defined as such, it's much safer to have pure functions that create new objects as needed rather than mutate objects they receive as arguments, possibly affecting the logic of other functions.

The difference here is that we're building the object, so we're in the case where indeed the type should be writeable. Ideally we'd used a writeable type/interface definition, and use `Readonly<T>` in the function arguments so we can have the readonly type checks.

An alternative would be to use the `Writable` type from `type-fest` when building the object: https://github.com/sindresorhus/type-fest/blob/main/source/writable.d.ts

Reply:
ah yeah, agreed about arguments passed to functions, I was thinking about this case where we're just building up an object iteratively.  I think for this case simply removing `readonly` from the fields within the `IndexModelArgs` interface definition would suffice.  Then it should be up to the places that use specific instances of `IndexModelArgs` to decide whether the instance should be mutable or not.

Reply:
Yeah sounds good to me to remove the `readonly` flags unless @ukstv has concerns about it?
I think we can incrementally add `Readonly<T>` to functions arguments as needed.